Source/Core/Content/ContentSystem.cpp

```diff
@@ -682,6 +678,47 @@ bool ContentSystem::RenameContentItemFile(ContentItem* contentItem, StringParam
   return true;
 }
 
+bool ContentSystem::MoveContentItem(ContentItem* contentItem, ContentLibrary* targetLibrary)
+{
+    // Build Strings
+
+    // Get the original full path to the content item and meta file
+    String oldId = contentItem->UniqueFileId;
+    String contentFile = contentItem->GetFullPath();
+    String contentMetaFile = contentItem->GetMetaFilePath();
+    
+    String oldPath = FilePath::GetDirectoryPath(contentFile);
+    String newFullPath = FilePath::Combine(targetLibrary->SourcePath, contentItem->Filename);
+
+    ContentLibrary* currentLibrary = contentItem->mLibrary;
+
+    // Does a file with this name already exist?
+    // Does a file with the same name exist in the target library folder? If not, then we need to move it.
+
+    bool fileExistsAtNewDirectory = FileExists(newFullPath);
+    if (fileExistsAtNewDirectory && (currentLibrary->SourcePath != targetLibrary->SourcePath))
+    {
+        return false;
+    }
+    else if (!fileExistsAtNewDirectory)
+    {
+        // If the file does not exists in the directory that we are trying to move it to, we need to remove the meta file and move the file.
+        if (!contentItem->DeleteMetaFile())
+            PlasmaPrint("Unable to delete meta file for %s", contentItem->Filename);
+        MoveFile(newFullPath, contentItem->GetFullPath());
+    }
+
+    currentLibrary->ContentItems.Erase(oldId);
+    contentItem->UniqueFileId = UniqueFileId(newFullPath);
+    contentItem->mLibrary = targetLibrary;
+    targetLibrary->ContentItems.Insert(contentItem->UniqueFileId, contentItem);
+
+    currentLibrary->Load();
```

Comment:
bumping this for visibility

Reply:
because the only contents that we have to save looks like this in the file itself

```
[Version:1]
ContentLibrary 
{
	var Name = "Easter_Eggs"
	var LibraryId = 0x60b02b662e8dcb13
	var SourceControlType = ""
	var ReadOnly = false
}

```

Reply:
Truthfully looking at this, the load function doesn't even need to be fired here